text/0000-kpack-donation-to-cnb.md

```diff
@@ -0,0 +1,351 @@
+```
+
+The _spec_ key is used to define the desired state of the [Image](https://github.com/pivotal/kpack/blob/main/docs/image.md)
+
+- _tag_: The image tag.
+- _builder_: Configuration of the [builder](https://github.com/pivotal/kpack/blob/main/docs/builders.md) resource the image builds will use.
+- source: The source code that will be monitored/built into images.
+
+# Migration
+
+### Repositories
+
+The suggested strategy for migrating [kpack's](https://github.com/pivotal/kpack/)  git repositories to the [CNB](https://buildpacks.io/) is to use the [transfer repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#transferring-a-repository-owned-by-your-organization) git feature.
+
+The following table shows the candidates repositories to be transferred
+
+| Origin Repo | Description | Owner | Destination Repo | Owner |
+| --- | --- | --- | --- | --- |
+| [https://github.com/pivotal/kpack](https://github.com/pivotal/kpack) | kpack source code | Pivotal | [https://github.com/buildpacks/kpack](https://github.com/buildpacks/kpack) | [CNB Technical Oversight Committee](https://github.com/buildpacks/community/blob/main/GOVERNANCE.md#technical-oversight-committee) |
+| [https://github.com/vmware-tanzu/kpack-cli](https://github.com/vmware-tanzu/kpack-cli) | kpack CLI | VMware | [https://github.com/buildpacks/kpack-cli](https://github.com/buildpacks/kpack-cli) | [CNB Technical Oversight Committee](https://github.com/buildpacks/community/blob/main/GOVERNANCE.md#technical-oversight-committee) |
+| [https://github.com/vmware-tanzu/homebrew-kpack-cli](https://github.com/vmware-tanzu/homebrew-kpack-cli) | Homebrew tap for the kpack CLI | VMware | [https://github.com/buildpacks/homebrew-kpack-cli](https://github.com/buildpacks/homebrew-kpack-cli) | [CNB Technical Oversight Committee](https://github.com/buildpacks/community/blob/main/GOVERNANCE.md#technical-oversight-committee) |
+
+For each repository
+
+- The owner or admin user must follow the steps describe in github [documentation](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#transferring-a-repository-owned-by-your-organization) and transfer the repository to the organization [Cloud Native Buildpacks](https://github.com/buildpacks)
+- A member of the [TOC team](https://github.com/orgs/buildpacks/teams/toc/members) in [CNB](https://buildpacks.io/) must accept the donation of the repository. The name of the destination repository will be the one described in the table above.
+
+### CI / CD Pipelines
+
+[kpack's](https://github.com/pivotal/kpack/) CI/CD infrastructure currently runs on internal VMware infrastructure. [kpack's](https://github.com/pivotal/kpack/) CI/CD pipelines will need to be rebuilt on [CNB](https://buildpacks.io/) infrastructure.
+
+##### Hardware requirements
+
+The minimal hardware requirements to request to CNCF to recreate the CI/CD pipelines are:
+
+- TBD
```

Comment:
@matthewmcnew Could you give me a hand here?

Reply:
3 nodes
100 GB ephemeral storage per node
8GB Memory per node
2 cpu per node

public gcr (eventually GAR) for registry (start with 100GB expected)

Reply:
@tylerphelan Thanks man!

Reply:
Can we have a bit more details on the kind of nodes we need?
Linux? Windows? amd64? arm?

Reply:
Good point. Here is my recommendation for the build component of the CI/CD pipeline:
Run builds  in a kpack installation on a K8s Cluster
- Linux Nodes
  - 3 Nodes (this pretty conservative)
  - amd64
  - 100GB ephemeral disk (this is also conservative)
  - 8GB memory
  - 2 cpu
- Windows (these are what we currently use to build the kpack windows images)
  - 1 node
  - amd64
  - 100GB ephemeral disk (this is because windows base images can be large)
  - 16GB (below this, pods would fail to pull windows builder images successfully)
  - 4 cpu
I don't think the control plane node resources matter as much, they can be low I assume.

The rest of the CI/CD pipeline could be in github actions free tier I believe.

Reply:
Wouldn't we need a cluster to test kpack as well? Likely a small linux only cluster

Reply:
@tomkennedy513 maybe the same build cluster and we can add another node ?

Reply:
Other k8s based projects just bring up a test cluster inside github actions. Would that be feasible?